sources/api/thar-be-updates/src/status.rs

```diff
@@ -260,22 +273,25 @@ impl UpdateStatus {
 
     /// Checks the list of updates to for an available update.
     /// If the 'version-lock'ed version is available returns true. Otherwise returns false
-    pub fn update_available_updates(
+    pub fn update_available_updates<P>(
         &mut self,
-        socket_path: &str,
+        config_path: P,
         updates: Vec<update_metadata::Update>,
-    ) -> Result<bool> {
+    ) -> Result<bool>
+    where
+        P: AsRef<Path>,
+    {
         // Extract the version to store
         self.available_updates = updates.iter().map(|u| u.version.to_owned()).collect();
         // Check if the 'version-lock'ed update is available as the 'chosen' update
         // Retrieve the 'version-lock' setting
-        let settings = get_settings(socket_path)?;
-        let locked_version: FriendlyVersion = serde_json::from_value(
-            settings["updates"]["version-lock"].to_owned(),
-        )
-        .context(error::GetSettingSnafu {
-            setting: "/settings/updates/version-lock",
-        })?;
+        let settings = get_settings(config_path)?;
+        let locked_version: FriendlyVersion =
+            settings
+                .version_lock
+                .context(error::GetSettingOptionSnafu {
+                    setting: "/settings/updates/version-lock",
+                })?;
```

Comment:
It looks like the only setting we need in this program is `version-lock`.

It might be better to just put that one setting in `/etc/thar-be-updates.toml` so it's clear what is actually used.

```
[required-extensions]
updates = "v1"
std = { version = "v1", helpers = ["default"] }
+++
version-lock = {{default "latest" settings.updates.version-lock}}
```